src/utilities/plugin-utils.test.ts

```diff
@@ -0,0 +1,200 @@
+        id: 0,
+        loaded: false,
+        plugin
+      }]);
+    });
+
+    it("does not add a used plugin when it is already in the list", () => {
+      expect(getUsedPlugins().length).toEqual(0);
+      addUsedPlugin(plugin);
+      expect(getUsedPlugins().length).toEqual(1);
+      addUsedPlugin(plugin);
+      expect(getUsedPlugins().length).toEqual(1);
+    });
+  });
+
+  describe("#findUsedPlugins", () => {
+    it("works in teacher edition mode", () => {
+      const usedPlugins = findUsedPlugins(activity, true);
+      expect(usedPlugins.length).toBe(3);
+    });
+
+    it("works in non teacher edition mode", () => {
+      const usedPlugins = findUsedPlugins(activity, false);
+      expect(usedPlugins.length).toBe(3);
+    });
+  });
+
+  describe("#loadPluginScripts", () => {
+    const MockLARA: LaraGlobalType = {
+      Plugins: {
+        setNextPluginLabel: jest.fn()
+      }
+    } as any;
+    const handleLoadPlugins = jest.fn();
+    let savedAppendChild: any;
+
+    beforeEach(() => {
+      jest.resetAllMocks();
+      savedAppendChild = document.body.appendChild;
+      document.body.appendChild = jest.fn((script: HTMLScriptElement) => {
+        (script.onload as any)();
+      }) as any;
+    });
+
+    afterEach(() => {
+      document.body.appendChild = savedAppendChild;
+    });
+
+    it("handles teacher edition mode", () => {
+      loadPluginScripts(MockLARA, activity, handleLoadPlugins, true);
+      expect(MockLARA.Plugins.setNextPluginLabel).toHaveBeenCalledTimes(3);
+      expect(handleLoadPlugins).toHaveBeenCalledTimes(1);
+    });
+
+    it("handles non teacher edition mode", () => {
+      loadPluginScripts(MockLARA, activity, handleLoadPlugins, false);
+      expect(MockLARA.Plugins.setNextPluginLabel).toHaveBeenCalledTimes(3);
+      expect(handleLoadPlugins).toHaveBeenCalledTimes(3);
+    });
+
+  });
+
+  describe("#validateEmbeddablePluginContextForPlugin", () => {
+    it("works when all the elements exist", () => {
+      const result = validateEmbeddablePluginContextForPlugin(partialContext);
+      expect(result?.LARA).toEqual(partialContext.LARA);
+      expect(result?.embeddable).toEqual(partialContext.embeddable);
+      expect(result?.embeddableContainer).toEqual(partialContext.embeddableContainer);
+    });
+  });
+
+  describe("#validateEmbeddablePluginContextForWrappedEmbeddable", () => {
+    it("works when all the wrapped elements exist", () => {
+      const result = validateEmbeddablePluginContextForWrappedEmbeddable(partialContext);
+      expect(result?.LARA).toEqual(partialContext.LARA);
+      expect(result?.embeddable).toEqual(partialContext.embeddable);
+      expect(result?.embeddableContainer).toEqual(partialContext.embeddableContainer);
+    });
+  });
+
+  describe("#loadLearnerPluginState", () => {
+    it("returns the state for all plugins in teacher mode", async () => {
+      const pluginState = await loadLearnerPluginState(activity, true);
```

Comment:
How does this work since there is no firebase connection? It seems like this should make a firebase connection to look for the state, is that being mocked somehow?

Reply:
This has been fixed in this commit (https://github.com/concord-consortium/activity-player/pull/132/commits/353170d182a5f9e1a4424f704d414f1910b02cb2).  There were multiple issues:

1. The lack of firebase connection caused the method to exit before setting the cache.
2. The lack of a return on the await in `loadLearnerPluginState` caused `null` to be returned.